File: decript.py
import string

def mostrar_banner():
    banner = r"""
╔════════════════════════════════════════════════════════════════╗   ║
║                                                                ║
║   🏛️  Descifrado César Aleatorio — Vigenère Style             ║
║   🧠  La clave revela lo oculto. El César sonríe.              ║
║   Script creado por: GsusH 🕷️🕸️🧙🏻‍♂️                   ║
╚════════════════════════════════════════════════════════════════╝
"""
    print(banner)

def descifrar_vigenere_aleatorio(texto_cifrado, clave_numerica):
    texto_cifrado = texto_cifrado.upper()
    alfabeto = string.ascii_uppercase
    texto_descifrado = ""

    for i, letra in enumerate(texto_cifrado):
        if letra in alfabeto:
            desplazamiento = clave_numerica[i]
            indice_cifrado = alfabeto.index(letra)
            indice_original = (indice_cifrado - desplazamiento) % 26
            texto_descifrado += alfabeto[indice_original]
        else:
            texto_descifrado += letra

    return texto_descifrado

# Ejecución
mostrar_banner()
mensaje_cifrado = input("Introduce el mensaje cifrado: ")
clave_str = input("Introduce la clave numérica separada por comas (ej. 3,12,5,...): ")
clave_numerica = [int(x.strip()) for x in clave_str.split(",")]

mensaje_original = descifrar_vigenere_aleatorio(mensaje_cifrado, clave_numerica)
print("\n📜 Mensaje original:", mensaje_original)

File: Scriptable-IOS/decript.js
// ===========================================
// DESCIFRADO VIGENÈRE ALEATORIO DESDE ATAJOS  
// Creado por GsusH 🕷️🕸️🧙🏻‍♂️
// ===========================================

// Entrada esperada: texto cifrado + clave numérica separados por "||"
// Ejemplo: "ZCWS QXJHR || 18,14,11,18,0,5,10,3,7,17"

let entrada = args.shortcutParameter;
let mensajeSalida = "";

if (!entrada || !entrada.includes("||")) {
  mensajeSalida = "❌ Entrada inválida. Usa el formato: texto || clave";
  Script.complete();
  return mensajeSalida;
}

// Separar texto cifrado y clave
let [textoCifrado, claveStr] = entrada.split("||").map(x => x.trim());

if (!textoCifrado || !claveStr) {
  mensajeSalida = "❌ Faltan datos. Asegúrate de incluir texto y clave separados por '||'.";
  Script.complete();
  return mensajeSalida;
}

let claveNumerica = claveStr.split(",").map(x => parseInt(x.trim()));
let alfabeto = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
let textoDescifrado = "";

textoCifrado = textoCifrado.toUpperCase();

for (let i = 0; i < textoCifrado.length; i++) {
  let letra = textoCifrado[i];
  let desplazamiento = claveNumerica[i] || 0;

  if (alfabeto.includes(letra)) {
    let indice = alfabeto.indexOf(letra);
    let original = alfabeto[(indice - desplazamiento + 26) % 26];
    textoDescifrado += original;
  } else {
    textoDescifrado += letra;
  }
}

mensajeSalida = `
📜 Mensaje original:
${textoDescifrado}
`;

Script.complete();
return mensajeSalida;
